### PR TITLE
Call out the State Flag for Custom Database/Schema Users

### PR DESCRIPTION
**Pull request by sungchun12**
Make it really easy to recommend using the `--state` flag for complex dbt setups

### CODE REVIEW

**Comment by sungchun12** [comment]
@dlawin ready for review!

File: docs/development_testing/open_source.md [hunk 1]
```diff
@@ -113,19 +113,9 @@ Run `pip install data-diff -U` to update!
 :::
 
 **There are two options for configuring data-diff --dbt:**
-1. Add variables to the dbt_project.yml
+1. Add variables to the `dbt_project.yml`
 2. Specify a production manifest.json using `--state`
```
Comment (leoebfolsom) [inline docs/development_testing/open_source.md:118]: If we're getting rid of the tabs (and I have no objection to that), I think we should have H4 (####) headers corresponding with the 2 options (dbt_project vs state). Maybe even link from each options (1 and 2) to the sections using anchors.

File: docs/development_testing/open_source.md [hunk 2]
```diff
@@ -185,18 +175,18 @@ Add the following variables to **dbt_project.yml**:
...
+**Using the `--state` option is highly recommended for dbt projects with multiple target database and schema configurations. For example, if you customized the [`generate_schema_name`](https://docs.getdbt.com/docs/build/custom-schemas#understanding-custom-schemas) macro, this is the best option for you.**
+
+> Note: `dbt ls` is preferred over `dbt compile` as it runs faster and data diffing does not require fully compiled models to work.
 
 ```bash
-  dbt compile -t prod # compile a manifest.json using the "prod" target
+  dbt ls -t prod # compile a manifest.json using the "prod" target
   mv target/manifest.json prod_manifest.json # move the file up a directory and rename it to prod_manifest.json
-  dbt compile # compile a manifest.json using a non "prod" target
+  dbt run # run your dbt models
```
Comment (leoebfolsom) [inline docs/development_testing/open_source.md:186]: Should we say something like "`dbt run` your entire project, or only the subset of models that will be diffed"?

File: docs/development_testing/open_source.md (hunk 1 above)
Comment (dlawin) [reply docs/development_testing/open_source.md:118]: agree

File: docs/development_testing/open_source.md (hunk 2 above)
Comment (sungchun12) [reply docs/development_testing/open_source.md]: updated for your feedback

File: docs/development_testing/open_source.md (hunk 1 above)
Comment (sungchun12) [reply docs/development_testing/open_source.md:117]: updated for the header sections

**Review by leoebfolsom — APPROVED** [review]

**Pull request merged**